### ISSUE TITLE
Clarify classpath dependency

### ISSUE THREAD

**Issue by carl-mastrangelo** [melix/jmh-gradle-plugin#70]
Hi, I work on [grpc](https://github.com/grpc/grpc-java) and we are using this gradle plugin for benchmarks.  It seems there is a discrepancy between the example on the README:
```
  dependencies {
    classpath "me.champeau.gradle:jmh-gradle-plugin:0.3.0"
  }
```
and the gradle page:  https://plugins.gradle.org/plugin/me.champeau.gradle.jmh

```
  dependencies {
    classpath "gradle.plugin.me.champeau.gradle:jmh-gradle-plugin:0.3.0"
  }
```

Which one is correct, and should they be made the same?

**Comment by vyazelenko** [comment]
@carl-mastrangelo Actually both are correct. README shows this:
```gradle
buildscript {
  repositories {
    jcenter()
  }
  dependencies {
    classpath "me.champeau.gradle:jmh-gradle-plugin:0.3.0"
  }
}

apply plugin: "me.champeau.gradle.jmh"
```
Which is correct since generated artifact has `groupId=me.champeau.gradle` and `artifactId=jmh-gradle-plugin`. And we are trying to download plugin from Bintray (`jcenter()`). _Btw release `0.3.0` **was not** published to Bintray._ 

Example shown in [plugins portal](https://plugins.gradle.org/plugin/me.champeau.gradle.jmh), i.e.:
```gradle
buildscript {
  repositories {
    maven {
      url "https://plugins.gradle.org/m2/"
    }
  }
  dependencies {
    classpath "gradle.plugin.me.champeau.gradle:jmh-gradle-plugin:0.3.0"
  }
}

apply plugin: "me.champeau.gradle.jmh"
```
Is also correct as it shows you how to download plugin directly from special plugins repo. It seems that in this repo artifact has `groupId=gradle.plugin.me.champeau.gradle`.

The easiest and 100% correct way to apply this plugin is to use new plugins syntax, i.e.:
```gradle
plugins {
  id "me.champeau.gradle.jmh" version "0.3.0"
}
```

/cc @aalmiray, @melix we need to ensure that plugin is always published to Bintray as well as to plugins portal.

**Comment by aalmiray** [comment]
Yes. We need the same groupId on both repositories in order to avoid confusion.

**Comment by ddimtirov** [comment]
@vyazelenko the new syntax [does not work if you need to use an internal artifact repository](https://discuss.gradle.org/t/is-there-a-way-to-proxy-the-gradle-plugin-portal-through-local-artifact-repository/6933) (i.e. for compliance reasons). This is a blocker for many teams.

 

It also [can not be used to apply plugins to subprojects](https://docs.gradle.org/current/userguide/plugins.html#N12EB3), forcing repetitions in the build file (though that has workarounds).

**Comment by aalmiray** [comment]
PR #74 fixes these problems

**Comment by MartinanderssonDotcom** [comment]
lol I can't get ANY syntax to work using the one in README or one of those found in issue threads. Gradle keep complaining about various kinds of dependency resolution errors.
